core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -48,29 +48,31 @@ interface LanguageInfo {
   templateUrl: './top-navigation-bar.component.html',
 })
 export class TopNavigationBarComponent implements OnInit, OnDestroy {
-  @Input() headerText: string;
-  @Input() subheaderText: string;
-
-  url: URL;
-  currentLanguageCode: string;
-  currentLanguageText: string;
-  isModerator: boolean;
-  isCurriculumAdmin: boolean;
-  isTopicManager: boolean;
-  isSuperAdmin: boolean;
-  isBlogAdmin: boolean;
-  isBlogPostEditor: boolean;
-  userIsLoggedIn: boolean;
-  username: string;
-  currentUrl: string;
-  userMenuIsShown: boolean;
-  inClassroomPage: boolean;
-  showLanguageSelector: boolean;
-  standardNavIsShown: boolean;
-  supportedSiteLanguages: LanguageInfo[];
-  ACTION_OPEN: string;
-  ACTION_CLOSE: string;
-  KEYBOARD_EVENT_TO_KEY_CODES: {
+  // The properties below are initialized in Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() headerText!: string;
+  @Input() subheaderText!: string;
+
+  url!: URL;
+  currentLanguageCode!: string;
+  supportedSiteLanguages!: LanguageInfo[];
+  currentLanguageText!: string;
+  isModerator!: boolean;
+  isCurriculumAdmin!: boolean;
+  isTopicManager!: boolean;
```

Comment:
initialize all properties having boolean types with a false value instead of non-null assertion

Reply:
I've initialized them with a false value.

However, this change also requires one of the tests (should get user information on initialization) to be changed. (For example I had to change `expect(component.isModerator).toBe(undefined);` to `expect(component.isModerator).toBe(false);` in the test)

I've had to make 5 such changes in the test. PTAL at the commit if you wish to inspect those changes!

Reply:
@mridul-netizen

Reply:
Yup, looks good....`undefined` value for a boolean property doesn't really makes sense it would've still be false.